__tests__/app.test.js

```diff
@@ -162,7 +162,83 @@ describe('GET /api/articles/:article_id/comments', () => {
         })
     })
   })
-  
+
+
+describe("POST: /api/articles/2/comments", () => {
+    test("status 201: should post a comment for an article and respond with the posted comment", () => {
+        const data = {
+            body: "This is a great article",
+            username: "butter_bridge"
+        }
+        return request(app)
+        .post('/api/articles/2/comments')
+        .send(data)
+        .expect(201)
+        .then(({ body }) => {
+            expect(body.comment).toMatchObject({
+                article_id:expect.any(Number),
+                author:expect.any(String),
+                body:expect.any(String),
+                comment_id:expect.any(Number),
+                created_at:expect.any(String),
+                votes:expect.any(Number)
+            })
+        })
+    })
+    test("status 400: no username inputted results in error message", () => {
+        const data = {
+            body: "This is a great article"
+        }
+        return request(app)
+        .post('/api/articles/2/comments')
+        .send(data)
+        .expect(400)
+        .then(({ body }) => {
+            expect(body.msg).toBe("Request is missing key information")
+        })
+    })
+    test("status 400: incorrect data type used for username results in error message", () => {
+        const data = {
+            username: 123,
+            body: "Great article"
+        }
+        return request(app)
+        .post('/api/articles/2/comments')
+        .send(data)
+        .expect(400)
+        .then(({ body }) => {
+            expect(body.msg).toBe("One or more inputs incorrectly formatted in request")
+        })
+    })
+    test("status 400: no body inputted results in error message", () => {
+        const data = {
+            username: "butter_bridge"
+        }
+        return request(app)
+        .post('/api/articles/2/comments')
+        .send(data)
+        .expect(400)
+        .then(({ body }) => {
+            expect(body.msg).toBe("Request is missing key information")
+        })
+    })
+    test("status 400: valid article id but no corresponding article", () => {
+        const data = {
+            body: "Wow fantastic",
+            username: "butter_bridge"
+        }
+        return request(app)
+        .post('/api/articles/9999/comments')
```

Comment:
Is this message - "one or more inputs incorrectly formatted in request" true for this test? the id is of the correct type - a number - and the comment is correctly formatted too. Therefore is 400 the appropriate status? 

As the article_id is valid just is _not found_ and the form of the comment being sent is correct there is a more appropriate error status and message, I reckon. 🙂

Reply:
Hey Mezz! I thought it was supposed to be a 404 but when I ran the test the error came back as the same PSQL error (23503) which I got when inputting the incorrect information which I'd already assigned the "400: one or more inputs incorrectly formatted in request" error to - is there a way to assign two different responses if the PSQL error is the same?